### PR TITLE
Feat: Playlist management

### PR DESCRIPTION
**Pull request by Rohitprasad83**
#  Added playlist management
- add a new playlist
- delete a playlist
- get all playlists 
- add a video to a playlist
- delete a video from a playlist
- get all videos from a playlist
 ---
DK4XVEMBXm

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -234,6 +235,58 @@
...
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+}
+
+.playlist-container-heading {
+    padding: 4rem 10rem 2rem 10rem;
+}
+
 @media (max-width: 768px) {
```
Comment (ishitakeshawani) [inline src/App.css:290]: You can use em here

File: src/components/videocard/VideoCard.jsx [hunk 2]
```diff
@@ -16,7 +25,38 @@ export function VideoCard({ video }) {
...
+  const [playlistTitle, setPlaylistTitle] = useState(false)
+  const [playlistDescription, setPlaylistDescription] = useState(false)
+  const [playlistId, setPlaylistId] = useState('')
+  const [playlists, setPlaylists] = useState([])
+
+  const location = useLocation()
+  const { encodedToken } = useAuth()
+
+  useEffect(() => {
+    ;(async () => {
```
Comment (ishitakeshawani) [inline src/components/videocard/VideoCard.jsx:39]: remove ; it from here

File: src/components/videocard/VideoCard.jsx [hunk 3]
```diff
@@ -16,7 +25,38 @@ export function VideoCard({ video }) {
...
+      try {
+        const response = await axios.get(`/api/user/playlists`, {
+          headers: {
+            authorization: encodedToken,
+          },
+        })
+        setPlaylists(response.data.playlists)
+      } catch (error) {
+        console.log(error)
```
Comment (ishitakeshawani) [inline src/components/videocard/VideoCard.jsx:48]: You can use toast to show error

File: src/components/videocard/VideoCard.jsx [hunk 4]
```diff
@@ -16,7 +25,38 @@ export function VideoCard({ video }) {
...
+      }
+    })()
+  }, [playlists])
+
+  useEffect(() => {
+    const path = location.pathname
+    if (path && path.includes('/playlist/')) {
+      setPlaylistId(path.substr(10))
```
Comment (ishitakeshawani) [inline src/components/videocard/VideoCard.jsx:56]: may be here you can also use useParams to get id from path.

**Review by ishitakeshawani — APPROVED** [review]
Added few suggestions. 
Code is readable.
If you add any Screenshots or preview link in PR then it will be better.

File: src/App.css [hunk 5]
```diff
@@ -234,6 +235,58 @@
     border-radius: 0.5rem;
```
Comment (omkarpatke) [inline src/App.css:235]: suggestion: use px if the unit is less than 1rem

File: src/App.css [hunk 6]
```diff
@@ -234,6 +235,58 @@
...
+}
+
+.createPlaylist {
+    display: flex;
+    flex-direction: column;
+}
+
+.createPlaylist>input {
+    margin: 1rem 0;
+    border: 2px solid var(--gray);
```
Comment (omkarpatke) [inline src/App.css:274]: Appreciation: good consistent use of color variables👍

**Review by omkarpatke — APPROVED** [review]
UI Looks Very good👍
code is also very well readable.
good and consistent use of CSS variables.

File: src/App.css (hunk 6 above)
Comment (Rohitprasad83) [reply src/App.css:274]: Thanks

File: src/components/videocard/VideoCard.jsx (hunk 4 above)
Comment (Rohitprasad83) [reply src/components/videocard/VideoCard.jsx:56]: Tried but was not working

File: src/components/videocard/VideoCard.jsx (hunk 3 above)
Comment (Rohitprasad83) [reply src/components/videocard/VideoCard.jsx:48]: will do it

File: src/components/videocard/VideoCard.jsx (hunk 2 above)
Comment (Rohitprasad83) [reply src/components/videocard/VideoCard.jsx:39]: sure

**Pull request merged**